libs/auth/scripts/src/generate_machine_label_qr_code_contents.ts

```diff
@@ -0,0 +1,124 @@
+import { Readable } from 'node:stream';
+import yargs from 'yargs/yargs';
+import { extractErrorMessage } from '@votingworks/basics';
+
+import { signMessage } from '../../src/cryptography';
+import { FileKey } from '../../src/keys';
+import { constructPrefixedMessage } from '../../src/signatures';
+
+interface CommandLineArgs {
+  privateKeyPath: string;
+  privateKeyPassphraseFilePath?: string;
+  serialNumber: string;
+}
+
+async function parseCommandLineArgs(
+  args: readonly string[]
+): Promise<CommandLineArgs> {
+  const argParser = yargs()
+    .options({
+      'private-key-path': {
+        description:
+          'The path to the VotingWorks label QR codes private key. ' +
+          'This is not the same key as our root private key.',
+        type: 'string',
+      },
+      'private-key-passphrase-file-path': {
+        description:
+          'The path to the passphrase file for the VotingWorks label QR codes private key. ' +
+          'If not provided, the passphrase will have to be entered interactively.',
+        type: 'string',
+      },
+      'serial-number': {
+        description:
+          'The machine serial number, also referred to as the machine ID',
+        type: 'string',
+      },
+    })
+    .hide('help')
+    .version(false)
+    .example('$ generate-machine-label-qr-code-contents --help', '')
+    .example(
+      '$ generate-machine-label-qr-code-contents \\\n' +
+        '--private-key-path path/to/private-key.pem \\\n' +
+        '--serial-number SC-00-000',
+      ''
+    )
+    .example(
+      '$ generate-machine-label-qr-code-contents \\\n' +
+        '--private-key-path path/to/private-key.pem \\\n' +
+        '--private-key-passphrase-file-path path/to/passphrase.txt \\\n' +
+        '--serial-number SC-00-000',
+      ''
+    )
+    .strict();
+
+  const helpMessage = await argParser.getHelp();
+  argParser.fail((errorMessage: string) => {
+    throw new Error(`${errorMessage}\n\n${helpMessage}`);
+  });
+
+  const parsedArgs = argParser.parse(args) as {
+    help?: boolean;
+    serialNumber?: string;
+    privateKeyPassphraseFilePath?: string;
+    privateKeyPath?: string;
+  };
+
+  if (parsedArgs.help) {
+    console.log(helpMessage);
+    process.exit(0);
+  }
+
+  if (!parsedArgs.serialNumber || !parsedArgs.privateKeyPath) {
+    console.error(helpMessage);
+    process.exit(1);
+  }
+
+  return {
+    serialNumber: parsedArgs.serialNumber,
+    privateKeyPassphraseFilePath: parsedArgs.privateKeyPassphraseFilePath,
+    privateKeyPath: parsedArgs.privateKeyPath,
+  };
+}
+
+async function generateMachineLabelQrCodeContents({
+  privateKeyPath,
+  privateKeyPassphraseFilePath,
+  serialNumber,
+}: CommandLineArgs): Promise<void> {
+  const signingPrivateKey: FileKey = {
+    source: 'file',
+    path: privateKeyPath,
+    passphraseFilePath: privateKeyPassphraseFilePath,
+  };
+  const message = constructPrefixedMessage(
+    'signed-serial-number',
+    `sn=${serialNumber}`
```

Comment:
let's assert a serial number format that is only chars, digits, and dashes.

Reply:
https://github.com/votingworks/vxsuite/pull/5768/commits/75f00c9a9c55d8d5944dffc194f0be19d3859070